lib/types/AbstractBuilder.js

```diff
@@ -39,20 +51,27 @@ class AbstractBuilder {
 	 * @param {array} tasksToRun List of tasks which should be executed
 	 * @returns {Promise} Returns promise chain with tasks
 	 */
-	build(tasksToRun) {
-		const allTasksCount = tasksToRun.filter((value) => this.availableTasks.includes(value)).length;
+	build() {
+		// Run phases and add tasks to queue
+		this.preprocess();
+		this.themebuilding();
+
+		if (!this.buildOptions.basic) {
+			this.process();
+			this.bundle();
+			this.optimize();
+		}
+
+		const allTasksCount = Object.keys(this.tasks).length;
 		this.taskLog.addWork(allTasksCount);
 
 		let taskChain = Promise.resolve();
-		for (let i = 0; i < this.availableTasks.length; i++) {
-			let taskName = this.availableTasks[i];
-
-			if (!tasksToRun.includes(taskName)) {
-				continue;
-			}
 
-			taskChain = taskChain.then(this.wrapTask(taskName, this.tasks[taskName]));
+		for (var i = 0; i < this.tasks.length; i++) {
```

Comment:
Move this into a separate function to allow easier overwrite of the `build` function